draft-ietf-jsonpath-base.md

```diff
@@ -1104,6 +1094,62 @@ Some examples:
 | `[3,4,5]` | `$[?index(@)==2]`<br>`$[?index(@)==17]` | `[5]`<br>`[]` | Select array element |
 | `{"a":{"b":{5},c:0}}` | `$[?@.b==5 && !@.c]` | `[{"b":{5},c:0}]` | Existence  |
 
+
+## Normalized Paths
+
+A Normalized Path identifies a node by providing a query that results in exactly that node. For example,
+the JSONPath expression `$.book[?(@.price<10)]` could select two values with Normalized Paths
+`$['book'][3]` and `$['book'][5]`.
+
+An example application of Normalized Paths is a JSONPath implementation which outputs Normalized Paths
+instead of, or in addition to, the values identified by these paths.
+
+Since bracket notation is more general than dot notation, it is used to construct Normalized Paths.
+
+Single quotes are used to delimit string member names. This reduces the number of characters which
+need escaping when Normalized Paths appear as strings (which are delimited with double quotes) in JSON documents.
+
+The syntax of Normalized Paths is restricted so that there is one and only one way of representing any
+given Normalized Path. Certain characters are escaped in one and only one way; all other
+characters are unescaped.
+
+~~~~ abnf
+normalized-path           = root-selector *(S normal-index-selector)
+normal-index-selector     = "[" S (normal-quoted-member-name / normal-element-index) S "]"
+normal-quoted-member-name = %x27 *normal-single-quoted %x27 ; 'string'
+normal-single-quoted      = normal-unescaped /
+                            ESC normal-escapable
+normal-unescaped          = %x20-26 /                       ; omit control codes
+                            %x28-5B /                       ; omit '
+                            %x5D-10FFFF                     ; omit \
+normal-escapable          = ( %x62 / %x66 / %x6E / %x72 / %x74 / ; \b \f \n \r \t
+                                ; b /         ;  BS backspace U+0008
+                                ; t /         ;  HT horizontal tab U+0009
+                                ; n /         ;  LF line feed U+000A
+                                ; f /         ;  FF form feed U+000C
+                                ; r /         ;  CR carriage return U+000D
+                                "'" /         ;  ' apostrophe U+0027
+                                "\" /         ;  \ backslash (reverse solidus) U+005C
+                                (%x75 normal-hexchar) ;  certain values u00XX U+00XX
+                            )
+normal-hexchar            = "0" "0"
+                            (
+                              ("0" "0") /
```

Comment:
This could be shortened as ("0" %x30-37)

Reply:
Agreed, but I think the longer form may be easier to read, so I'm going to punt on this for now.